test/groovy/DockerExecuteTest.groovy

```diff
@@ -1,61 +1,97 @@
 
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
 
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import util.JenkinsConfigRule
+import util.JenkinsLoggingRule
+import util.JenkinsSetupRule
+
 import static org.junit.Assert.assertEquals
 import static org.junit.Assert.assertTrue
 import static org.junit.Assert.assertFalse
 
-class DockerExecuteTest extends PiperTestBase {
+class DockerExecuteTest extends BasePipelineTest {
     private DockerMock docker
 
-    String echos
+    @Rule
+    public JenkinsSetupRule jsr = new JenkinsSetupRule(this)
+
+    @Rule
+    public JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
+
+    @Rule
+    public JenkinsConfigRule jcr = new JenkinsConfigRule(this)
```

Comment:
we should maybe use a RuleChain in all tests where multiple rules are used?

Reply:
done